### PR TITLE
Make sure loading screens use dark mode user preference.

### PR DESCRIPTION
**Pull request by MitjaBezensek**
We load the user preferences a bit earlier, so that we can make sure that the `LoadingScreen` and `ErrorScreen` also use the correct color and background color based on the dark mode setting.

There's still a brief flash of white screen, but that's before any of our components load, not sure if we can avoid that one.

Solves https://github.com/tldraw/tldraw/issues/1248

### Change Type

- [x] `patch` — Bug Fix

### Test Plan

1. Probably best if you throttle your network speed.
2. Reload the page.
3. The asset loading screen should use take your dark mode setting into account.
4. Change the dark mode and try again.

### Release Notes

- Make sure our loading and error screens take dark mode setting into account.

### CODE REVIEW

**Comment by steveruizok** [comment]
This looks good to me, afaik the editor should support creating an editor like this just fine. @ds300 could you confirm?

File: packages/editor/src/lib/TldrawEditor.tsx [hunk 1]
```diff
@@ -156,13 +165,20 @@ function TldrawEditorWithOwnStore(
...
 	store,
+	user,
 	...rest
-}: Required<TldrawEditorProps & { store: TLStoreWithStatus }, 'shapes' | 'tools'>) {
+}: Required<TldrawEditorProps & { store: TLStoreWithStatus; user: TLUser }, 'shapes' | 'tools'>) {
+	const container = useContainer()
+	if (user.userPreferences.value.isDarkMode) {
+		container.classList.remove('tl-theme__light')
+		container.classList.add('tl-theme__dark')
+	}
```
Comment (ds300) [inline packages/editor/src/lib/TldrawEditor.tsx:180]: this should probably be done in a useLayoutEffect

**Review by ds300 — APPROVED** [review]
Seems fine to me aside from doing a DOM mutation during a render. I think that should be moved into a useLayoutEffect.

File: packages/editor/src/lib/TldrawEditor.tsx (hunk 1 above)
Comment (steveruizok) [reply packages/editor/src/lib/TldrawEditor.tsx:180]: taken this one

**Pull request merged**